pkg/sql/opt/norm/rules/join.opt

```diff
@@ -532,6 +532,28 @@ $left
     $private
 )
 
+#ConvertProjectionsToFiltersLeft
+[ConvertProjectionsToFiltersLeft, Normalize]
+(InnerJoin | InnerJoinApply | LeftJoin | LeftJoinApply | SemiJoin
+    $left:(Project)
+    $right:*
+    $on: * & (CanExtractTupleEqualityLeft $left $right $on)
+    $private:*
+)
+=>
+(ExtractTupleEqualityLeft (OpName) $left $right $on $private)
```

Comment:
Similarly, we try to minimize what goes into the functions. It would be much better if we could write the "replace" side of the rule along the lines of `(Project (Join ... (ExtractTupleEq ...)) $projections (OutputCols2 .. .. )`

Reply:
Again - definitely simplifies things - changes made and commited